models/User.php

```diff
@@ -990,6 +990,28 @@ public function getKeyBindings()
         return $this->keyBindings ? $this->keyBindings : self::getDefaultKeyBindings();
     }
 
+    /**
+     * @param array|null $bindings
+     * @return array|null
+     */
+    public static function strictKeybinds($bindings)
+    {
+        if($bindings == null){
+            return null;
+        }
+        foreach($bindings as $ind => $binding){
+            if(!array_key_exists('ctrl', $binding)){
+                $bindings[$ind]['ctrl'] = false;
+            }
+            if(!array_key_exists('alt', $binding)){
+                $bindings[$ind]['alt'] = false;
+            }
+            if(!array_key_exists('shift', $binding)){
+                $bindings[$ind]['shift'] = false;
+            }
+        }
```

Comment:
Fancy modern code:

```suggestion
        foreach (array_keys($bindings) as $idx) {
            $bindings[$idx]['ctrl'] ??= false;
            $bindings[$idx]['alt'] ??= false;
            $bindings[$idx]['shift'] ??= false;
        }

```

The difference is, that `null` will be turned to `false` while it stays `null` in your solution. Does that matter?

Reply:
just checked it out and this will still work great, thanks!